ts/torch_handler/base_handler.py

```diff
@@ -183,9 +250,14 @@ def inference(self, data, *args, **kwargs):
         Returns:
             Torch Tensor : The Predicted Torch Tensor is returned in this function.
         """
-        marshalled_data = data.to(self.device)
         with torch.no_grad():
-            results = self.model(marshalled_data, *args, **kwargs)
+            if hasattr(self.model, "run"):
+                data = data.to(torch.float32).cpu().numpy()
+                # TODO: Should we make this "modelInput configurable", feels complicated
+                results = self.model.run(None, {"modelInput": data})[0]
```

Comment:
I think we should leave this part to user, and have documents that ask user to implement the inference or maybe a utility like

```
output =[node.name for node in model.graph.output]

input_all = [node.name for node in model.graph.input]
input_initializer =  [node.name for node in model.graph.initializer]
input = list(set(input_all)  - set(input_initializer))
```
may help to then assert if the passed input matches the expected ones?

Reply:
Yeah interesting idea, lemme take a look at this

Reply:
Another idea would be to not run the inference at all in the base handler

I can recommend users create their own inference function where the inference happens so that the scope of ONNX support really means initializing the model